src/dialect/mssql/mssql-driver.ts

```diff
@@ -238,6 +242,19 @@ class MssqlConnection implements DatabaseConnection {
     return tediousIsolationLevel
   }
 
+  #cancelRequest<O>(request: MssqlRequest<O>): Promise<void> {
+    return new Promise((resolve) => {
+      request.request.once('requestCompleted', resolve)
+
+      const wasCanceled = this.#connection.cancel()
```

Comment:
this is `true` when the underlying `tedious` request has completed already - all rows were received from database.